docs/proposals/scheduling/binding-priority-preemption/README.md

```diff
@@ -0,0 +1,340 @@
+
+I submit a AI workload which occupies preferentially GPU resources of member clusters.
+Karmada tries to find feasible clusters that can run the AI workload. If no enough member clusters are found,
+I hope Karmada tries to remove bindings with lower priority from some arbitrary member clusters in order to make room for the pending bindings.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+Binding priority and preemption can have unwanted side effects. Here is an example of potential problems and ways to deal with them.
+
+#### Bindings are preempted unnecessarily
+
+Preemption removes existing bindings from some member clusters under resource pressure to make room for higher priority pending bindings.
+If you give high priorities to certain bindings by mistake, these unintentionally high priority bindings may cause preemption in your multi-cluster system.
+
+To address the problem, you can disable the preemption behavior for your bindings.
+
+When a binding is preempted, there will be events recorded for the preempted binding.
+Preemption should happen only when a member cluster does not have enough resources for a binding.
+In such cases, preemption happens only when the priority of the pending binding (preemptor) is higher than the preempted bindings.
+Preemption must not happen when there is no pending binding, or when the pending bindings have equal or lower priority than the preempted bindings.
+If preemption happens in such scenarios, please fell free to file an issue.
+
+## Design Details
+
+### Effect of priority on scheduling
+
+One could generally expect a binding with higher priority has a higher chance of getting scheduled than the same binding with lower priority.
+However, there are other parameters that affect scheduling decisions. So, a high priority binding may or may not be scheduled before lower priority bindings.
+For example, a high priority binding is marked as unschedulable due to insufficient resources of member clusters, but the preemption behavior is not enabled.
+Karmada will try to schedule others with lower priority.
+
+### Effect of priority on preemption
+
+Generally, lower priority bindings are more likely to get preempted by higher priority bindings when member clusters have reached a threshold.
+In such a case, scheduler may decide to preempt lower priority bindings to release enough resources for higher priority pending bindings.
+As mentioned before, there are other parameters that affect scheduling decisions, such as cluster affinity and spread constraints.
+If scheduler determines that a high priority binding cannot be scheduled even if lower priority bindings are preempted, it will not preempt lower priority bindings.
+
+### New API
+
+#### Priority Classes
+
+This proposal are going to add a new API object called `PriorityClass`. The priority class defines the mapping between the priority name and its value.
```

Comment:
Do you mean to use the `scheduling.k8s.io/v1 PriorityClass` or introduce a brand new PriorityClass to Karmada API?

In my opinion, we can share the one from Kubernetes and don't have to introduce a new one.

@LeonZh0u [mentioned the concept of `federated priority class`](https://docs.google.com/document/d/1MixmgLwnmiRrukyFP25JvE2hqiqB3f_JKT12tXiOsbc/edit?disco=AAABUpFnU68). You both have similar ideas, so we can clarify them here.

Reply:
In order to facilitate the expansion of these APIs in the future, we'd better introduce a brand new PriorityClass to Karmada API.

Reply:
we are following @whitewindmills proposal's federated priority class. and will be ready for a demo on the priority workflow in our next community meeting.

Reply:
@LeonZh0u Please help to add an agenda the [Meeting Notes](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit).

@whitewindmills will you attend the next meeting? Probably next Tuesday 23:00 PM UTC+8.

Reply:
sorry, I may not be able to attend.